### PR TITLE
[MSYS-769] Win32-certstore - get certificate

### PR DESCRIPTION
**Pull request by piyushawasthi**
**Note:** First merge PR: #16 after that need to rebase this branch.

**Description:**
This method get certificate using thumbprint and return OpenSSL::X509 certificate object.

**Usages:**
```
thumbprint = "‎75 e0 ab b6 13 85 12 27 1c 04 f8 5f dd de 38 e4 b7 24 2e fe"
Win32::Certstore.open("ROOT") do |store|
  cert_obj = store.get("‎75 e0 ab b6 13 85 12 27 1c 04 f8 5f dd de 38 e4 b7 24 2e fe")
  store.close
end
```
**Note:**  Here cert_obj is OpenSSL::X509 certificate object as.
```
#<OpenSSL::X509::Certificate
 subject=#<OpenSSL::X509::Name CN=GlobalSign Root CA,OU=Root CA,O=GlobalSign nv-sa,C=BE>,
 issuer=#<OpenSSL::X509::Name CN=GlobalSign Root CA,OU=Root CA,O=GlobalSign nv-sa,C=BE>,
 serial=#<OpenSSL::BN 4835703278459707669005204>,
 not_before=1998-09-01 12:00:00 UTC,
 not_after=2028-01-28 12:00:00 UTC>
```

### CODE REVIEW

**Comment by piyushawasthi** [comment]
This branch is rebase and ready to merge.

File: lib/win32/certstore/mixin/helper.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#
+# Author:: Piyush Awasthi (<piyush.awasthi@msystechnologies.com>)
+# Copyright:: Copyright (c) 2017 Chef Software, Inc.
```
Comment (btm) [inline lib/win32/certstore/mixin/helper.rb]: nitpick, but check these as it is 2018 now.

File: lib/win32/certstore/store_base.rb [hunk 2]
```diff
@@ -125,11 +111,30 @@ def cert_get_name_args(pcert_context, cert_name)
...
+      end
+
       # Convert OpenSSL::X509::Certificate object in .der formate
       def der_cert(cert_obj)
         FFI::MemoryPointer.from_string(cert_obj.to_der)
       end
 
+      # Get certificate pem
+      def get_cert_pem(thumbprint)
+        IO::popen(["powershell.exe", cert_ps_cmd(thumbprint)]) {|io| io.readlines}
```
Comment (btm) [inline lib/win32/certstore/store_base.rb]: ideally we do what we need with a ruby library, and then with FFI if there is no library, and finally a shell-out as a last resort. Is there a reason you're not using mixlib-shellout here? We've put a lot of work into shellout to make sure we can capture things like stdin, stdout, error codes, etc. reliably in different circumstances,

File: spec/win32/unit/certstore_spec.rb [hunk 3]
```diff
@@ -98,6 +98,95 @@
     end
   end
 
+  describe "#cert_get" do
+    context "When passing empty certificate store name" do
+      let (:store_name) { "" }
+      it "raises ArgumentError" do
+        expect { certstore.open(store_name) }.to raise_error("Invalid Certificate Store.")
```
Comment (btm) [inline spec/win32/unit/certstore_spec.rb]: we should check the exception type here, i.e. `.to raise_error(ArgumentError, "Invalid Certificate Store")`

File: spec/win32/unit/certstore_spec.rb [hunk 4]
```diff
@@ -98,6 +98,95 @@
...
+      let (:store_name) { "root" }
+      let (:thumbprint) { "b1 bc 96 8b d4 f4 9d 62 2a a8 9a 81 f2 15 01 52 a4 1d 82 9c" }
+      let (:cert_pem) { File.read('.\spec\win32\unit\assets\GlobalSignRootCA.pem') }
+      before(:each) do
+        allow_any_instance_of(certbase).to receive(:get_cert_pem).and_return([cert_pem])
+      end
+      it "returns OpenSSL::X509::Certificate Object" do
+        store = certstore.open(store_name)
+        cert_obj = store.get(thumbprint)
+        expect(cert_obj.class.to_s).to eql("OpenSSL::X509::Certificate")
```
Comment (btm) [inline spec/win32/unit/certstore_spec.rb]: `expect(cert_obj).to be_an_instance_of(OpenSSL::X509::Certificate)`

File: lib/win32/certstore/store_base.rb [hunk 5]
```diff
@@ -125,11 +111,30 @@ def cert_get_name_args(pcert_context, cert_name)
...
+      end
+
+      # Format pem
+      def format_pem(cert_pem)
+        cert_pem.join
+      end
+
+      # Build pem to OpenSSL::X509::Certificate object
+      def build_openssl_obj(cert_pem)
+        OpenSSL::X509::Certificate.new cert_pem
```
Comment (btm) [inline lib/win32/certstore/store_base.rb]: `OpenSSL::X509::Certificate.new(cert_pem)`

https://github.com/bbatsov/ruby-style-guide#method-invocation-parens

Comment (piyushawasthi) [reply lib/win32/certstore/store_base.rb]: sorry for that.

**Comment by piyushawasthi** [comment]
@btm : Thanks a lot for review I have fixed all comments, Please have a look and let me know in case of any concerns.

**Pull request merged**